### PR TITLE
[risk=low][RW-7616] Add the current DUCC version(s) to config and allow multiple

### PR DESCRIPTION
**Pull request by jmthibault79**
Add the current DUCC version to config instead of hardcoding.

We will need to support multiple "current" DUCC versions for RW-7616.  This PR doesn't change our requirement, but it sets us up to do so.

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/config/config_local.json [hunk 1]
```diff
@@ -98,7 +98,8 @@
     "unsafeAllowSelfBypass": true,
     "unsafeAllowUserCreationFromGSuiteData": true,
     "enableRasLoginGovLinking": true,
-    "enforceRasLoginGovLinking": true
+    "enforceRasLoginGovLinking": true,
+    "currentDuccVersions": [3]
```
Comment (jmthibault79) [inline api/config/config_local.json:102]: See AccessModuleServiceImpl.CURRENT_DATA_USER_CODE_OF_CONDUCT_VERSION

File: api/src/test/java/org/pmiops/workbench/api/ProfileControllerTest.java [hunk 2]
```diff
@@ -142,20 +140,17 @@
...
   @Autowired private InstitutionService institutionService;
   @Autowired private ProfileController profileController;
   @Autowired private ProfileService profileService;
   @Autowired private UserDao userDao;
   @Autowired private UserCodeOfConductAgreementDao userCodeOfConductAgreementDao;
-  @Autowired private UserDataUseAgreementDao userDataUseAgreementDao;
   @Autowired private UserTermsOfServiceDao userTermsOfServiceDao;
   @Autowired private FakeClock fakeClock;
 
-  // allows us to mock out only specific methods
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/api/ProfileControllerTest.java:156]: no longer needed - can be autowired now

File: api/src/test/java/org/pmiops/workbench/api/ProfileControllerTest.java [hunk 3]
```diff
@@ -142,20 +140,17 @@
...
   @Autowired private AccessModuleDao accessModuleDao;
+  @Autowired private AccessModuleService accessModuleService;
   @Autowired private AccessTierDao accessTierDao;
   @Autowired private AccessTierService accessTierService;
   @Autowired private InstitutionService institutionService;
   @Autowired private ProfileController profileController;
   @Autowired private ProfileService profileService;
   @Autowired private UserDao userDao;
   @Autowired private UserCodeOfConductAgreementDao userCodeOfConductAgreementDao;
-  @Autowired private UserDataUseAgreementDao userDataUseAgreementDao;
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/api/ProfileControllerTest.java:152]: unused after recent changes

File: api/src/test/java/org/pmiops/workbench/db/dao/UserServiceTest.java [hunk 4]
```diff
@@ -442,20 +443,25 @@ public void testSubmitTermsOfService() {
...
-
-    verify(accessModuleService, never()).updateCompletionTime(any(), any(), any());
+  public void testSyncDuccVersionStatus_correctVersions() {
+    providedWorkbenchConfig.access.currentDuccVersions = ImmutableList.of(3, 4, 5);
+
+    providedWorkbenchConfig.access.currentDuccVersions.forEach(
+        version -> {
+          final DbUser user = userDao.findUserByUsername(USERNAME);
+          user.setDuccAgreement(signDucc(user, version));
+          userDao.save(user);
```
Comment (freemabd) [inline api/src/test/java/org/pmiops/workbench/db/dao/UserServiceTest.java:453]:
```suggestion
          DbUser user = userDao.findUserByUsername(USERNAME);
          user.setDuccAgreement(signDucc(user, version));
          user = userDao.save(user);
```
Use the returned instance for further operations as the save operation might have changed the entity instance completely

**Review by freemabd — APPROVED** [review]
LGTM

**Pull request merged**